includes/wikia/api/ArticlesApiController.class.php

```diff
@@ -1048,6 +1056,73 @@ public function getAsJson() {
 		$this->setResponseData( $result, '', self::SIMPLE_JSON_VARNISH_CACHE_EXPIRATION );
 	}
 
+	/**
+	 * Get some or all of the article top level sections as an array
+	 * @param mixed $sectionsToGet Array of section numbers or the string "all"
+	 * @param ParserOutput $parsedArticle
+	 * @param Article $article
+	 * @param Title $title
+	 * @return array
+	 * @throws BadRequestApiException
+	 */
+	private function getArticleSections( $sectionsToGet, ParserOutput $parsedArticle, Article $article, $title ) {
+		$content = [];
+		$articleAsWikiText = $article->getContent();
+		$parsedSections = $parsedArticle->getSections();
+
+		if ( $sectionsToGet === 'all' || in_array( 'all', $sectionsToGet ) ) {
+			$sectionsToGet = range( 0, count( $parsedArticle->getSections() ) );
+		}
+		if ( is_array( $sectionsToGet ) ) {
+			foreach ( $sectionsToGet as $section ) {
+				// Only retrieve top level sections.
+				// $parsedSections doesn't include section 0, so subtract 1
+				$isTopLevel = $parsedSections[$section - 1]['toclevel'] === 1;
+				if ( $section == 0 || $isTopLevel ) {
+					$content[] = $this->getArticleSection( $articleAsWikiText, $title, $section );
+				}
+			}
+		} else {
+			throw new BadRequestApiException( 'Sections must be an array of numbers or "all"' );
+		}
+		return $content;
+	}
+
+	/**
+	 * Get the HTML for an article section
+	 * @param string $articleAsWikiText
+	 * @param Title $title
+	 * @param integer $section
+	 * @return string
+	 * @todo Supposedly getSection works with HTML, but I could only get it to work with wikitext
+	 */
+	private function getArticleSection( $articleAsWikiText, $title, $section ) {
+		/** @var $wgParser Parser */
```

Comment:
The datatype should come first here, and since its a single variable declaration, the variable name can actually be left off.